stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -6,15 +6,15 @@ eigencopy_oftype(A::Symmetric, S) = Symmetric(copy_similar(A, S), sym_uplo(A.upl
 
 # Eigensolvers for symmetric and Hermitian matrices
 eigen!(A::RealHermSymComplexHerm{<:BlasReal,<:StridedMatrix}; sortby::Union{Function,Nothing}=nothing) =
-    Eigen(sorteig!(LAPACK.syevr!('V', 'A', A.uplo, A.data, 0.0, 0.0, 0, 0, -1.0)..., sortby)...)
+    Eigen(sorteig!(LAPACK.syevr!('V', 'A', A.uplo, A.data, 0.0, 0.0, 0, 0, -1.0)..., sortby)..., true)
```

Comment:
A default value of `true`  in the constructor of `Eigen` would avoid the changes in this file.

Reply:
This is super dangerous, I'd rather the default case be the one that is correct in general

Reply:
Indeed, packages that don't set the optional argument will potentially run by the wrong code paths.

Reply:
For `Diagonal` matrices, which are normal, the value of the boolean field should be `true`, what is not the case currently. The same for `Bidiagonal` and others, when the off diagonals are zero.